templates/projects/macros.j2

```diff
@@ -8,7 +8,8 @@
   {% set client = page.config.client[0] if page.config.client else null %}
   {% set contributors = page.config.contributors if page.config.contributors else null %}
   {% set project = page.config.project[0] %}
-  {% set link = ['/', page.slug, '/']|join('') if project.owner or project.case_study %}
+  {% set slug = ('/' + page.slug|replace('index', '')) if ('index' in page.slug) else ('/' + page.slug + '/') %}
+  {% set link = slug if (project.owner or project.case_study) else none %}
```

Comment:
Yeah, we could maybe break out some of this logic that we reuse into either smaller macros with clear names, or actual filters.